static/docs/command-reference/checkout.md

```diff
@@ -17,59 +17,54 @@ positional arguments:
 ## Description
 
 [DVC-files](/doc/user-guide/dvc-file-format) in a <abbr>project</abbr> specify
-which instance of each data file or directory should be used, with the checksums
-saved in the `outs` field. The `dvc checkout` command updates the workspace data
-to match with the <abbr>cached</abbr> files that correspond to those checksums.
-
-Using an SCM like Git, the DVC-files are kept under version control. At a given
-branch or tag of the SCM repository, the DVC-files will contain checksums for
-the corresponding data files kept in the cache. After an SCM command like
-`git checkout` is run, the DVC-files will change to the state at the specified
-branch or commit or tag. Afterwards, the `dvc checkout` command is required in
-order to synchronize the data files with the currently checked out DVC-files.
-
-This command must be executed after `git checkout` since Git doesn't track files
-that are under DVC control. For convenience a Git hook is available, simply by
-running `dvc install`, that will automate running `dvc checkout` after
-`git checkout`. See `dvc install` for more information.
-
-The execution of `dvc checkout` does:
-
-- Scan the `outs` entries in DVC-files to compare with the currently checked out
-  data files. The scanned DVC-files is limited by the listed `targets` (if any)
-  on the command line. And if the `--with-deps` option is specified, it scans
-  backward from the given `targets` in the corresponding
-  [pipeline](/doc/command-reference/pipeline).
-
-- For any data files where the checksum doesn't match their DVC-file entry, the
-  data file is restored from the cache. The link strategy used (`reflink`,
-  `hardlink`, `symlink`, or `copy`) depends on the OS and the configured value
-  for `cache.type` – See `dvc config cache`.
-
-Note that this command by default tries NOT to copy files between the cache and
-the workspace, using reflinks instead when supported by the file system. (Refer
-to
+which data files or directories from the <abbr>cache</abbr> should be in use. We
+call these files <abbr>outputs</abbr>, and their checksums are saved in the
+`outs` fields inside DVC-files to achieve this.
+
+When using Git, different DVC-files versioned in separate
+[revisions](https://git-scm.com/book/en/v2/Git-Internals-Git-References)
+probably specify different data files from the cache. When switching to those
+versions (with Git commands such as `git checkout`), the current DVC-files will
+no longer match with all of the data in the <abbr>workspace</abbr>.
+
+The `dvc checkout` command synchronizes the workspace data to match with the
```

Comment:
introduction is way too long before we actually say anything about `dvc checkout` and what it does. Also, it's too tight with `git checkout` this way. It's usual case for `dvc checkout` but not a single one (e.g. `git clone` + `dvc checkout` is a legit one). The point is - the first paragraph should describe the essence of the operation - read DVC-files + restore the workspace accordingly. Then we can start talking about details - when it's useful, why, the logic behind it (mention different modes, etc).

Reply:
I would also try to use "files and directories" vs outputs terminology as much as possible.

Reply:
OK I simplified the description intro into 2 paragraphs which start by mentioning what `dvc checkout` does.

Not sure why we want to avoid the "output" term. I see it as a good thing when we get to use our terminology in general (without forcing it). But OK, I can rewrite it.

Reply:
Still a bit too complicated :)

"DVC-files can be considered as a pointer or a placeholder for the actual data file or a directory under DVC control. This command reads DVC-files and updates DVC-controlled data files and directories in the workspace to be in sync with the versions specified in the DVC-files." - something along those lines?

Reply:
I think it it also make sense to mention that files to checkout come from cache (not remote, the difference between checkout and pull, for example). It's also important since a few paragraphs below we talk about cache and it's not clear why.

Reply:
I simplified it again along the lines of your suggestion, thanks. There's still a short 2nd paragraph emphasizing that this is useful in the context of Git repos.

> it also make sense to mention that files to checkout come from cache

It is already mentioned in the 2nd bullet about the execution of the command, before the other mentions of the cache you saw. I did wrap it with `<abbr>` now though, so it's more obvious and has a tooltip.